src/providers/SubscriptionProvider/SubscriptionProvider.actions.js

```diff
@@ -12,6 +13,33 @@ import {
   ACTIVE
 } from './SubscriptionProvider.constants';
 
+export function updateIsOnTrialPeriod() {
+  return (dispatch, getState) => {
+    const userCreatedAt = getState().app.userData.createdAt;
+    const { isSubscribed } = getState().subscription;
+    const isOnTrialPeriod = isUserOnTrialPeriod(userCreatedAt);
+    dispatch({
+      type: UPDATE_IS_ON_TRIAL_PERIOD,
+      isOnTrialPeriod
+    });
+
+    if (!isOnTrialPeriod && !isSubscribed)
+      dispatch(showPremiumRequired({ showTryPeriodFinishedMessages: true }));
+
+    function isUserOnTrialPeriod(createdAt) {
+      const createdAtDate = new Date(createdAt);
+      const actualDate = new Date();
+      const DAYS_TO_TRY = 30;
+      const tryLimitDate = createdAtDate.setDate(
+        createdAtDate.getDate() + DAYS_TO_TRY
+      );
+      if (!createdAt) return false; //this case are already created users
```

Comment:
Move this condition to line 30 to avoid unnecessary process throughout the function